File: tests/fixtures/sample.go
// Test fixture: Go sample code for scanner/parser/search tests.

package main

import (
	"errors"
	"fmt"
)

// User represents a system user.
type User struct {
	ID       int
	Name     string
	Email    string
	IsActive bool
}

// UserRepository provides access to user storage.
type UserRepository interface {
	FindByID(id int) (*User, error)
	FindByEmail(email string) (*User, error)
	Save(user *User) error
	Delete(id int) error
}

// AuthService handles user authentication.
type AuthService struct {
	repo UserRepository
}

// NewAuthService creates a new authentication service.
func NewAuthService(repo UserRepository) *AuthService {
	return &AuthService{repo: repo}
}

// Authenticate validates user credentials and returns a token.
func (s *AuthService) Authenticate(email, password string) (string, error) {
	user, err := s.repo.FindByEmail(email)
	if err != nil {
		return "", fmt.Errorf("authentication failed: %w", err)
	}
	if user == nil {
		return "", errors.New("user not found")
	}
	// TODO: verify password hash
	return fmt.Sprintf("token-%d", user.ID), nil
}

// CalculateAverage computes the average of a slice of numbers.
func CalculateAverage(numbers []float64) (float64, error) {
	if len(numbers) == 0 {
		return 0, errors.New("cannot calculate average of empty slice")
	}
	sum := 0.0
	for _, n := range numbers {
		sum += n
	}
	return sum / float64(len(numbers)), nil
}

File: tests/fixtures/sample.rs
// Test fixture: Rust sample code for scanner/parser/search tests.

/// A user authentication handler.
///
/// Validates the provided credentials against the database
/// and returns a session token on success.
pub fn authenticate_user(username: &str, password: &str) -> Result<String, AuthError> {
    let user = find_user_by_name(username)?;
    if verify_password(password, &user.password_hash) {
        Ok(generate_session_token(&user))
    } else {
        Err(AuthError::InvalidCredentials)
    }
}

/// Calculate the total price of items in a shopping cart.
pub fn calculate_total(items: &[CartItem]) -> f64 {
    items.iter().map(|item| item.price * item.quantity as f64).sum()
}

/// A simple struct representing a user.
#[derive(Debug, Clone)]
pub struct User {
    pub id: u64,
    pub name: String,
    pub email: String,
    pub password_hash: String,
}

/// Error types for authentication.
#[derive(Debug)]
pub enum AuthError {
    UserNotFound,
    InvalidCredentials,
    DatabaseError(String),
}

struct CartItem {
    name: String,
    price: f64,
    quantity: u32,
}

fn find_user_by_name(_username: &str) -> Result<User, AuthError> {
    todo!()
}

fn verify_password(_password: &str, _hash: &str) -> bool {
    todo!()
}

fn generate_session_token(_user: &User) -> String {
    todo!()
}
